src/RequestHandler/inprocess/inprocessapplication.cpp

```diff
@@ -544,27 +562,82 @@ IN_PROCESS_APPLICATION::ExecuteApplication(
 
-    RunDotnetApplication(m_pConfig->QueryHostFxrArgCount(), m_pConfig->QueryHostFxrArguments(), pProc);
+    hr = RunDotnetApplication(m_pConfig->QueryHostFxrArgCount(), m_pConfig->QueryHostFxrArguments(), pProc);
 
 Finished:
     //
     // this method is called by the background thread and should never exit unless shutdown
     //
-    if (!m_fRecycleProcessCalled)
+    if (!m_fRecycleProcessCalled && FAILED(hr))
     {
-        STRU                    strEventMsg;
+        // Close the pipe as Read will hang if it isn't set.
+        CHAR pzFileContents[4096] = { 0 };
+        DWORD dwNumBytesRead;
+        STRU  struStdErrLog;
+        LARGE_INTEGER   li = { 0 };
+
+        // Try to read std err logs for application exception.
+        // If anything fails here, ignore it.
+
+        if (m_pConfig->QueryStdoutLogEnabled())
+        {
+            if (m_hLogFileHandle != INVALID_HANDLE_VALUE)
+            {
+                if (!GetFileSizeEx(m_hLogFileHandle, &li))
+                {
+                    goto Finished;
+                }
+
+                if (li.LowPart > 0)
+                {
+                    DWORD dwFilePointer = SetFilePointer(m_hLogFileHandle, 0, NULL, FILE_BEGIN);
+                    if (dwFilePointer != INVALID_SET_FILE_POINTER)
+                    {
+                        if (ReadFile(m_hLogFileHandle, pzFileContents, 4096, &dwNumBytesRead, NULL))
+                        {
+                            struStdErrLog.CopyA(pzFileContents, dwNumBytesRead);
+                        }
+                    }
+                }
+            }
+        }
+        else
+        {
+            if (m_hErrReadPipe != INVALID_HANDLE_VALUE)
+            {
+                // Close the write pipe before reading such that read file will not hang
+                // waiting for data.
+                if (m_hErrWritePipe != INVALID_HANDLE_VALUE)
+                {
+                    CloseHandle(m_hErrWritePipe);
+                }
+
+                DWORD dwFilePointer = SetFilePointer(m_hErrReadPipe, 0, NULL, FILE_BEGIN);
+                if (dwFilePointer != INVALID_SET_FILE_POINTER)
+                {
+                    if (ReadFile(m_hErrReadPipe, pzFileContents, 4096, &dwNumBytesRead, NULL))
+                    {
+                        struStdErrLog.CopyA(pzFileContents, dwNumBytesRead);
+                    }
+                }
+            }
+        }
+
+        STRU strEventMsg;
         if (SUCCEEDED(strEventMsg.SafeSnwprintf(
-                ASPNETCORE_EVENT_INPROCESS_THREAD_EXIT_MSG,
-                m_pConfig->QueryApplicationPath()->QueryStr(),
-                m_pConfig->QueryApplicationPhysicalPath()->QueryStr(),
-                m_ProcessExitCode)))
+            ASPNETCORE_EVENT_INPROCESS_THREAD_EXIT_MSG,
+            m_pConfig->QueryApplicationPath()->QueryStr(),
+            m_pConfig->QueryApplicationPhysicalPath()->QueryStr(),
+            hr,
+            struStdErrLog.QueryStr())))
```

Comment:
if struStdErrLog is empty, do we still want to log empty or hardcoded some info?